### DISCUSSION TITLE
Prevent approval on draft PRs

### DISCUSSION THREAD

**Discussion by jvincent-siplay** [community/community#10087] [Pull Requests]
I created a pull request and then realized I had more work to do, so I converted it to a draft.  One of my teammates approved the PR without realizing that I plan to make additional commits.  It seems like GitHub should not allow approvals on draft PRs until after we click "Ready for review."

It might also be good to have some more obvious indicator that a PR is in draft mode when looking at the Files tab.  Maybe a banner, or change the green "Review changes" button to a different color or something like that.

**Comment by salieflewis** [comment]
Bump. Is this possible?

**Comment by philippdieng** [comment]
Same happend to our team. I don't really understand why draft allows an approval. Especially when an approval has been committed you can click on "Ready for review" and then merge. I don't think that's the meaning of a draft.

Preventing from approving would be awesome!

**Comment by cmayorga** [comment]
Same here, some kind of disable approval even reviews for drafts would be nice

**Comment by sijanpaudel14** [comment]
I was also facing this issue

**Comment by swedishpotato** [comment]
It seems obvious to me. If a PR is draft it is not finished by definition. So how come someone else can review and merge it? It’s not always obvious a PR is in draft if people are too quick and have a lot of other stuff to work through.

**Reply by cat-cooperm** [reply]
> It seems obvious to me. If a PR is draft it is not finished by definition. So how come someone else can review and merge it? It’s not always obvious a PR is in draft if people are too quick and have a lot of other stuff to work through.

I've had to deal with this a few times. I can't really blame anyone in particular, especially if they are in team lead position and are trying to keep on top of many things going on, but a draft PR should simply not allow approvals to be given; only review comments.

**Comment by Otterpocket** [comment]
This allowed subversion of process really messes up the draft/ready functionality for me. Is there any benefits for allowing draft PR's to be reviewed? If not I would love to have this looked into.

**Comment by blangston** [comment]
I've found it useful to have draft PRs reviewed in the situation where the code is ready but should not yet be merged, since draft PRs prevent merging.

This happens, for instance, when code is complete but there are some separate infrastructure steps that need to happen first in order for the code to be deployable.  If deployment happens via automated CD processes on merge, then it's still nice to get a review of the code before the infra is in place.  Then, once missing infra pieces are in place, the draft PR can be converted to a regular PR.  At that point, it is unblocked from being merged/deployed.

Not saying this is necessarily the best procedure, but have found it useful in some cases.

**Reply by spullab** [reply]
Agree with @blangston, I frequently find myself in that position as someone who works in a different timezone from my approvers
